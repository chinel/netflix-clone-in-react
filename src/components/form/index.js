import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
