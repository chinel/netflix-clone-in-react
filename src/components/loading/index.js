import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
