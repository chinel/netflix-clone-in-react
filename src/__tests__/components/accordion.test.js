import React from "react";
import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
  it("renders the <Accordion /> with populated data", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        {faqsData.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>{item.body}</Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>
    );
    expect(getByText("Frequently Asked Questions")).toBeTruthy();
    expect(getByText("What is Netflix")).toBeTruthy();
    expect(getByText("How much does Netflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel")).toBeTruthy();
    expect(getByText("What can I watch on Netflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
