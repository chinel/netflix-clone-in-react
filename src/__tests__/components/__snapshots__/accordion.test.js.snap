// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion/> component 1`] = `
<div
  class="accordion__Container-z1197o-0 cWGcUO"
>
  <div
    class="accordion__Inner-z1197o-1 fQgJZ"
  >
    <h1
      class="accordion__Title-z1197o-5 geSyRJ"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-z1197o-0 cWGcUO"
>
  <div
    class="accordion__Inner-z1197o-1 fQgJZ"
  >
    <h1
      class="accordion__Title-z1197o-5 geSyRJ"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-z1197o-2 iGfNWJ"
    >
      <div
        class="accordion__Header-z1197o-3 jUfllx"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
